caas/kubernetes/clientconfig/k8s.go

```diff
@@ -18,12 +18,11 @@ var logger = loggo.GetLogger("juju.caas.kubernetes.clientconfig")
 // K8SClientConfig parses Kubernetes client configuration from the default location or $KUBECONFIG.
 func K8SClientConfig() (*ClientConfig, error) {
 
-	configPath := getKubeConfigPath()
-
-	config, err := clientcmd.LoadFromFile(configPath)
+	config, err := getKubeConfig()
 	if err != nil {
-		return nil, errors.Annotatef(err, "failed to read kubernetes config from '%s'", configPath)
+		return nil, err
```

Comment:
I am assuming that getKubeConfig() could be called from many places, so adding errors.Trace(err) here would be helpful \o/